### PR TITLE
DIFF tests are not printing error output

### PR DESCRIPTION
**Pull request by milljm**
All tests labeled as DIFF type tests are failing to output the diff to stdout.

Refs #8541

### CODE REVIEW

**Comment by brianmoose** [comment]
How about adding to the `testDiffs` test in `test_HarnessTester.py`.
Something like:
```
self.assertRegexpMatches(e.output, 'test_harness\.exodiff.*?Running exodiff')
self.assertRegexpMatches(e.output, 'test_harness\.csvdiff.*?Running CSVDiffer.py')
```
Or even something more specific if you want.

File: python/TestHarness/unit_tests/test_HarnessTester.py [hunk 1]
```diff
@@ -44,8 +44,8 @@ def testDiffs(self):
             self.runTests('-i', 'diffs')
 
         e = cm.exception
-        self.assertRegexpMatches(e.output, 'test_harness\.exodiff.*?FAILED \(EXODIFF\)')
-        self.assertRegexpMatches(e.output, 'test_harness\.csvdiff.*?FAILED \(CSVDIFF\)')
+        self.assertRegexpMatches(e.output, 'test_harness\.exodiff.*?Running exodiff')
+        self.assertRegexpMatches(e.output, 'test_harness\.csvdiff.*?Running CSVDiffer.py')
```
Comment (permcody) [inline python/TestHarness/unit_tests/test_HarnessTester.py]: You don't want to replace what's here or you'll break what this was testing. (e.g. The fact that it prints FAILED (EXODIFF)). Instead you'll want to add theses lines to check that it's recognizing some sort of content in the middle AND the final message that it failed. Nothings stopping you from having both RegExs.

**Review by permcody — CHANGES_REQUESTED** [review]
This looks good with the exception I noted.

**Review by permcody — APPROVED** [review]

File: python/TestHarness/testers/Exodiff.py [hunk 2]
```diff
@@ -51,7 +51,6 @@ def processResults(self, moose_dir, retcode, options, output):
         output = RunApp.processResults(self, moose_dir, retcode, options, output)
 
         if self.getStatus() == self.bucket_fail or self.specs['skip_checks']:
-            self.setStatus('skip checked', self.bucket_skip)
             return output
```
Comment (milljm) [inline python/TestHarness/testers/Exodiff.py]: Accidentally treated this as a 'skipped' test, when in fact this needs to be a failure. The status of this test would have already been set within RunApp's processResults method.

**Comment by permcody** [comment]
Testing this locally. Working as expected.

**Pull request merged**